src/helpers/map.js

```diff
@@ -169,3 +173,163 @@ export function findTile(tiles) {
   console.error(`Unable to find tile: ${tiles}`);
   return {};
 }
+
+
+/*
+ * Initializes and returns a 2D array with the specified width, height, and
+ * default value.
+ *
+ * width: the width of the initialized 2D array
+ * height: the height of the initialized 2D array
+ * defaultVal: the value to give each element in the initialized 2D array
+ */
+export function init2dArray(width, height, defaultVal) {
+  const matrix = [];
+
+  for (let x = 0; x < width; x++) {
+    matrix[x] = new Array(height);
+    for (let y = 0; y < height; y++) {
+      matrix[x][y] = defaultVal;
+    }
+  }
+
+  return matrix;
+}
+
+
+/*
+ * Returns the sum of all corresponding elements from two matrices being
+ * multiplied together. For example:
+ *   a = 1 2   b = 4 1
+ *       3 4       2 3
+ *   multiplyCorrespondingElementsAndSum(a, b)
+ *   (1*4) + (2*1) + (3*2) + (4*3) = 21
+ *
+ * m1: the first matrix
+ * m2: the second matrix (of the same dimensions as m1)
+ */
+export function multiplyCorrespondingElementsAndSum(m1, m2) {
+  let condition = (m1.length === m2.length);
+  let errorMessage = 'multiplyCorrespondingElementsAndSum: m1 and m2 are not the same width';
+  utils.assert(condition, errorMessage);
+
+  condition = (m1.length === m2.length);
+  errorMessage = 'multiplyCorrespondingElementsAndSum: m1 and m2 are not the same height';
+  utils.assert(condition, errorMessage);
```

Comment:
Oh god the asserts. This is so beautiful. But it must have taken you forever. ❤️